src/client/datascience/kernel-launcher/kernelEnvVarsService.ts

```diff
@@ -18,75 +17,79 @@ export class KernelEnvironmentVariablesService {
         @inject(IInterpreterService) private readonly interpreterService: IInterpreterService,
         @inject(IEnvironmentActivationService) private readonly envActivation: IEnvironmentActivationService,
         @inject(IEnvironmentVariablesService) private readonly envVarsService: IEnvironmentVariablesService,
-        @inject(IPlatformService) private readonly platformService: IPlatformService,
-        @inject(IEnvironmentVariablesProvider) private readonly customEndVars: IEnvironmentVariablesProvider
+        @inject(IEnvironmentVariablesProvider) private readonly customEnvVars: IEnvironmentVariablesProvider
     ) {}
     /**
      * If the kernel belongs to a conda environment, then use the env variables of the conda environment and merge that with the env variables of the kernel spec.
      * In the case of some kernels such as java, the kernel spec contains the cli as such `argv = ['java', 'xyz']`.
      * The first argument is an executable, and it is not in the current path.
      * However, when activating the conda env, the path variables are updated to set path to the location where the java executable is located.
      */
-    public async getEnvironmentVariables(resource: Resource, kernelSpec: IJupyterKernelSpec) {
-        const customEditVarsPromise = this.customEndVars.getCustomEnvironmentVariables(resource);
-        customEditVarsPromise.catch(noop); // If we return early, we don't want failing promise go unhandled.
+    public async getEnvironmentVariables(
+        resource: Resource,
+        interpreter: PythonEnvironment | undefined,
+        kernelSpec: IJupyterKernelSpec
+    ) {
         let kernelEnv = kernelSpec.env && Object.keys(kernelSpec.env).length > 0 ? kernelSpec.env : undefined;
-        if (!kernelSpec.interpreterPath) {
-            traceInfo(
-                `No custom variables for Kernel as interpreter path is not defined for kernel ${kernelSpec.display_name}`
-            );
-            return kernelEnv;
+
+        // If an interpreter was not explicitly passed in, check for an interpreter path in the kernelspec to use
+        if (!interpreter) {
```

Comment:
I kept the addition of the interpreter parameter on this function. For the case where we are launching with a known pythonEnv it seems better to use the actual launch interpreter that we will be using as opposed to the one in the kernelSpec interpreterPath. but added back the old code path for any case where the launch interpreter is not already calculated.